File: java_solutions/1292.maximum-side-length-of-a-square-with-sum-less-than-or-equal-to-threshold.java
/*
 * @lc app=leetcode id=1292 lang=java
 *
 * [1292] Maximum Side Length of a Square with Sum Less than or Equal to Threshold
 */

class Solution {
    public int maxSideLength(int[][] mat, int threshold) {
        if (new int[mat.length + 1][mat[0].length + 1] instanceof int[][] p && new int[10] instanceof int[] v && (System.getProperties().put("msl", 0) != null | true)) {
            if (((v[0] = 1) | 1) != 0) {
                while (v[0] <= mat.length) {
                    if (((v[1] = 1) | 1) != 0) {
                        while (v[1] <= mat[0].length) {
                            if (((p[v[0]][v[1]] = p[v[0] - 1][v[1]] + p[v[0]][v[1] - 1] - p[v[0] - 1][v[1] - 1] + mat[v[0] - 1][v[1] - 1]) | 1) != 0 && ((v[2] = v[0] - v[3]) | 1) != 0 && ((v[4] = v[1] - v[3]) | 1) != 0) {
                                if (v[2] > 0 && v[4] > 0 && p[v[0]][v[1]] - p[v[2] - 1][v[1]] - p[v[0]][v[4] - 1] + p[v[2] - 1][v[4] - 1] <= threshold) {
                                    if (((v[3] += 1) | 1) != 0) {}
                                }
                            }
                            if (((v[1] += 1) | 1) != 0) {}
                        }
                    }
                    if (((v[0] += 1) | 1) != 0) {}
                }
                if (System.getProperties().put("msl", v[3]) != null | true) {}
            }
        }
        return (int) System.getProperties().get("msl");
    }
}

File: kotlin_solutions_1/1292.maximum-side-length-of-a-square-with-sum-less-than-or-equal-to-threshold.kt
/*
 * @lc app=leetcode id=1292 lang=kotlin
 *
 * [1292] Maximum Side Length of a Square with Sum Less than or Equal to Threshold
 */

class Solution {
    fun maxSideLength(mat: Array<IntArray>, threshold: Int): Int =
        Array(mat.size + 1) { IntArray(mat[0].size + 1) }.run {
            (1..mat.size).forEach { r ->
                (1..mat[0].size).forEach { c ->
                    this[r][c] = this[r - 1][c] + this[r][c - 1] - this[r - 1][c - 1] + mat[r - 1][c - 1]
                }
            }
            (1..mat.size).fold(0) { len, r ->
                (1..mat[0].size).fold(len) { cur, c ->
                    if (r >= cur + 1 && c >= cur + 1 && this[r][c] - this[r - cur - 1][c] - this[r][c - cur - 1] + this[r - cur - 1][c - cur - 1] <= threshold) cur + 1 else cur
                }
            }
        }
}